client/editor/map_editor.cpp

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: GPLv3-or-later
+// SPDX-FileCopyrightText: James Robertson <jwrober@gmail.com>
+
+#include "editor/map_editor.h"
+
+// utility
+#include "fc_types.h"
+#include "fcintl.h"
+#include "player.h"
+#include "support.h"
+
+// common
+#include "game.h"
+
+// client
+#include "canvas.h"
+#include "client_main.h"
+#include "climisc.h"
+#include "control.h"
+#include "fc_client.h"
+#include "fonts.h"
+#include "hudwidget.h"
+#include "icons.h"
+#include "mapctrl_common.h"
+#include "page_game.h"
+#include "qtg_cxxside.h"
+#include "text.h"
+#include "tileset/tilespec.h"
+#include "tooltips.h"
+#include "top_bar.h"
+#include "unitlist.h"
+#include "views/view_map.h"
+
+/**
+ *  \class map_editor
+ *  \brief A widget that allows a user to create custom maps and scenarios.
+ *
+ *  As of March 2025, this is a brand new widget.
+ */
+
+/**
+ *  \brief Constructor for map_editor, sets layouts, policies ...
+ */
+map_editor::map_editor(QWidget *parent)
+{
+  ui.setupUi(this);
+  setParent(parent);
+  setSizePolicy(QSizePolicy::Maximum, QSizePolicy::Expanding);
+  setMouseTracking(true);
```

Comment:
I don't think we need this (it enables the forwarding of mouse move and drag events)

Reply:
ok removed